state-chain/pallets/cf-broadcast/src/lib.rs

```diff
@@ -1098,8 +1097,9 @@ impl<T: Config<I>, I: 'static> Broadcaster<T::TargetChain> for Pallet<T, I> {
 		AbortedBroadcasts::<T, I>::mutate(|aborted| {
 			aborted.remove(&broadcast_id);
 		});
-		let mut api_call = Self::clean_up_broadcast_storage(broadcast_id)
-			.ok_or(Error::<T, I>::ApiCallUnavailable)?;
+
+		let mut api_call =
```

Comment:
We don't delete here because it means that if we start to resign and then we succeed in witnessing it, the witness will not go through. We can leave the data here, and it will be overwritten at the appropriate time (on_signature_ready) if necessary.